specification/containerservice/resource-manager/Microsoft.ContainerService/aks/preview/2023-07-02-preview/managedClusters.json

```diff
@@ -0,0 +1,8779 @@
+              }
+            ]
+          },
+          "title": "The upgrade channel for auto upgrade. The default is 'none'.",
+          "description": "For more information see [setting the AKS cluster auto-upgrade channel](https://docs.microsoft.com/azure/aks/upgrade-cluster#set-auto-upgrade-channel)."
+        },
+        "nodeOSUpgradeChannel": {
+          "type": "string",
+          "enum": [
+            "Unmanaged",
+            "None",
+            "SecurityPatch",
+            "NodeImage"
+          ],
+          "x-ms-enum": {
+            "name": "NodeOSUpgradeChannel",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "None",
+                "description": "No attempt to update your machines OS will be made either by OS or by rolling VHDs. This means you are responsible for your security updates"
+              },
+              {
+                "value": "Unmanaged",
+                "description": "OS updates will be applied automatically through the OS built-in patching infrastructure. Newly scaled in machines will be unpatched initially, and will be patched at some later time by the OS's infrastructure. Behavior of this option depends on the OS in question. Ubuntu and Mariner apply security patches through unattended upgrade roughly once a day around 06:00 UTC. Windows does not apply security patches automatically and so for them this option is equivalent to None till further notice"
+              },
+              {
+                "value": "SecurityPatch",
+                "description": "AKS will update the nodes VHD with patches from the image maintainer labelled \"security only\" on a regular basis. Where possible, patches will also be applied without reimaging to existing nodes. Some patches, such as kernel patches, cannot be applied to existing nodes without disruption. For such patches, the VHD will be updated, and machines will be rolling reimaged to that VHD following maintenance windows and surge settings. This option incurs the extra cost of hosting the VHDs in your node resource group."
+              },
+              {
+                "value": "NodeImage",
+                "description": "AKS will update the nodes with a newly patched VHD containing security fixes and bugfixes on a weekly cadence. With the VHD update machines will be rolling reimaged to that VHD following maintenance windows and surge settings. No extra VHD cost is incurred when choosing this option as AKS hosts the images."
+              }
+            ]
+          },
+          "title": "Manner in which the OS on your nodes is updated",
+          "description": "The default is Unmanaged, but may change to either NodeImage or SecurityPatch at GA."
+        }
+      },
+      "description": "Auto upgrade profile for a managed cluster."
+    },
+    "UpgradeOverrideSettings": {
+      "type": "object",
+      "properties": {
+        "forceUpgrade": {
```

Comment:
Consider using enum instead:
https://github.com/Azure/azure-openapi-validator/blob/main/docs/enum-instead-of-boolean.md#why-the-rule-is-important

Reply:
This is designed to simplify the api. The force upgrade is a boolean that allows bypassing a few validations.

In the future, when we introduce single validation bypassing, we'll make that one a enum instead.